unit_tests/test_configuration.py

```diff
@@ -34,6 +36,46 @@ def test_bootstrap(self):
             FileContains(matcher=Contains("<numExecutors>1</numExecutors>")))
         self.assertEqual({8080}, self.fakes.juju.ports["TCP"])
 
+    def test_set_prefix(self):
+        test_file = tempfile.NamedTemporaryFile()
+        paths.DEFAULTS_CONFIG_FILE = test_file.name
+        test_file.write(b"# port for HTTP connector\nHTTP_PORT=8080\n")
+        test_file.write(b'JENKINS_ARGS="--httpPort=$HTTP_PORT"')
+        test_file.seek(0)
+
+        # Case #1 - no previous config, no prefix, no change
+        updated = self.configuration._set_prefix("")
+        self.assertFalse(updated)
+
+        # Case #2 - no previous config, a prefix, expected change
+        updated = self.configuration._set_prefix("/jenkins")
+        self.assertTrue(updated)
+
+        # Case #3 - previous config, same prefix, no expected change
+        updated = self.configuration._set_prefix("/jenkins")
+        self.assertFalse(updated)
+
+        # Case #4 - previous config, different prefix, expected change
+        updated = self.configuration._set_prefix("/jenkins-alt")
+        self.assertTrue(updated)
+
+        # Case #5 - previous config, no prefix, expected change
+        updated = self.configuration._set_prefix("")
+        self.assertTrue(updated)
+
+        # Case #6 - no config file, no expected change
```

Comment:
Please break down all these separate cases in separate test methods. You'll probably want a small helper function to create the test file.

In this way if something breaks you can have an instant look at all edge cases that don't pass, instead of having to run the suite N times and discover a new failure each time.